### ISSUE TITLE
camlp5-7.00 does not build with OCaml 4.01.0

### ISSUE THREAD

**Issue by vbgl** [camlp5/camlp5#9]
When attempting to build the newly released `camlp5` version 7.00 with OCaml 4.01.0, I get the following error:

```
ocamlc.opt -warn-error A -I ../odyl -I ../../boot -I ../../lib -I ../../ocaml_stuff/4.01.0/utils -I ../../ocaml_stuff/4.01.0/parsing -c -impl ast2pt.ppo
File "ast2pt.ml", line 302, characters 37-46:
Error: This function has type
         string list -> Parsetree.core_type -> Parsetree.core_type_desc
       It is applied to too many arguments; maybe you forgot a `;'.
make[3]: *** [../../config/Makefile:22: ast2pt.cmo] Error 2
```

Looking at the history of the `ast2pt.ml` file, it seems that commit [b00c2963b106eb8ef0ba53343ab99272cfeadfa3](https://github.com/camlp5/camlp5/commit/b00c2963b106eb8ef0ba53343ab99272cfeadfa3#diff-46797eb40d17837e79e1124ce62cb3c7L434) changed the function `ocaml_ptyp_poly` and gave to it a different signature depending on the OCaml version.

**Comment by roglo** [comment]
Fixed. You can use the Camlp5 git version (master). I plan to make a release but I have another bug to fix before.

**Comment by roglo** [comment]
Fixed. The Camlp5 github repository (master) is good now. 

Le 2017-06-26 18:16, vbgl a écrit :

> When attempting to build the newly released camlp5 version 7.00 with OCaml 4.01.0, I get the following error:
> 
> ocamlc.opt -warn-error A -I ../odyl -I ../../boot -I ../../lib -I ../../ocaml_stuff/4.01.0/utils -I ../../ocaml_stuff/4.01.0/parsing -c -impl ast2pt.ppo
> File "ast2pt.ml", line 302, characters 37-46:
> Error: This function has type
> string list -> Parsetree.core_type -> Parsetree.core_type_desc
> It is applied to too many arguments; maybe you forgot a `;'.
> make[3]: *** [../../config/Makefile:22: ast2pt.cmo] Error 2
> 
> Looking at the history of the ast2pt.ml file, it seems that commit b00c2963b106eb8ef0ba53343ab99272cfeadfa3 [1] changed the function ocaml_ptyp_poly and gave to it a different signature depending on the OCaml version. 
> 
> --
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub [2], or mute the thread [3].

  

Links:
------
[1]
https://github.com/camlp5/camlp5/commit/b00c2963b106eb8ef0ba53343ab99272cfeadfa3#diff-46797eb40d17837e79e1124ce62cb3c7L434
[2] https://github.com/camlp5/camlp5/issues/9
[3]
